src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -761,16 +762,15 @@ internal static TimeSpan TicksToTimeSpan(double ticks)
             return TimeSpan.FromSeconds(ticks / (double)ticksPerSecond);
         }
 
-        /// <summary>Opens a stream around the specified file descriptor and with the specified access.</summary>
-        /// <param name="fd">The file descriptor.</param>
+        /// <summary>Opens a stream around the specified socket file descriptor and with the specified access.</summary>
+        /// <param name="fd">The socket file descriptor.</param>
         /// <param name="access">The access mode.</param>
         /// <returns>The opened stream.</returns>
-        private static FileStream OpenStream(int fd, FileAccess access)
+        private static Stream OpenStream(int fd, FileAccess access)
         {
             Debug.Assert(fd >= 0);
-            return new FileStream(
-                new SafeFileHandle((IntPtr)fd, ownsHandle: true),
-                access, StreamBufferSize, isAsync: false);
+            var socket = new Socket(new SafeSocketHandle((IntPtr)fd, ownsHandle: true));
```

Comment:
On Mac, it's throwing here, because `Socket._isConnected` is apparently never set to `true` when initializing with the file descriptors from `Interop.Sys.ForkAndExecProcess()`

Reply:
Yeah, we infer _isConnected based on whether we were able to get a remote peer name... apparently that's not working well on macOS with socketpair.
https://github.com/dotnet/runtime/blob/dda0a5701ef6a3a4e66b223695016c2260c617e0/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L233

Reply:
I can take a look. I was planning to follow-up on #34727 anyway.

Reply:
Thanks!

Reply:
By definition, a `socketpair()` is always connected. Maybe `SafeSocketHandle` could have an internal constructor with a `bool assumeConnected` parameter?

Reply:
I'd rather just fix the implementation to correctly infer it.  Sounds like @wfurt's already done that. ;)